consensus/protocol.go

```diff
@@ -38,4 +39,7 @@ type Protocol interface {
 
 	// ValidateSemantic validates a block is correctly derived from its parent.
 	ValidateSemantic(ctx context.Context, child *types.Block, parents *types.TipSet) error
+
+	// BlockTime returns the block time used by the consensus protocol.
+	BlockTime(ctx context.Context) time.Duration
```

Comment:
I am unclear on where we draw the line on what methods require a `Context` and which don't, seems like all of them?

Reply:
Good point, the method returns a constant, I should not have added context as a parameter here, will remove.

Reply:
There is a case to be made that all interface methods that retrieve state should take a context. The context is primarily used to cleanly cancel IO calls and other code that's not entirely synchronous. So a method failing to take a context leaks implementation details. The same logic would require all methods to return an error. The general pattern on this project is thread through contexts when needed and leave them when the need goes away. So this context is ok but not strictly necessary.